src/QMCWaveFunctions/TWFFastDerivWrapper.h

```diff
@@ -57,6 +61,28 @@ class TWFFastDerivWrapper
   void addGroup(const ParticleSet& P, const IndexType groupid, SPOSet* spo);
   inline void addJastrow(WaveFunctionComponent* j) { jastrow_list_.push_back(j); };
 
+  /**
+   * @brief add a MultiSlaterDet
+   * 
+   * this is called when registering a MSD
+   * the caller should also iterate over MultiDiracDeterminants in the MSD and register the associated SPOSets
+   * 
+   * @param P. ParticleSet 
+   * @param msd. Pointer to MultiSlaterDetTableMethod
+   */
+  void addMultiSlaterDet(const ParticleSet& P, const WaveFunctionComponent* msd);
+  inline IndexType numMultiSlaterDets() const { return slaterdets_.size(); }
+  inline bool hasMultiSlaterDet() const
+  {
+    if (numMultiSlaterDets() > 1)
+    {
+      APP_ABORT("ERROR: TWFFastDerivWrapper does not support multiple MultiSlaterDetTableMethod");
```

Comment:
Please throw exception instead of using APP_ABORT. See https://qmcpack.readthedocs.io/en/develop/developing.html#log-and-error-output

Reply:
this one is gone with the change to only holding a single MSD, but I've added two more that are exceptions instead of APP_ABORT
1. if trying to getMultiSlaterDet() when one hasn't been registered
2. if trying to register a MSD in a wrapper that already has one